Chessington.GameEngine/Pieces/Piece.cs

```diff
@@ -20,5 +20,63 @@ public void MoveTo(Board board, Square newSquare)
             var currentSquare = board.FindPiece(this);
             board.MovePiece(currentSquare, newSquare);
         }
+
+        protected IEnumerable<Square> GetAvailableLateralMoves(Board board)
+        {
+            var zeroes = Enumerable.Repeat(0, 15);
+            var minusSevenToSeven = Enumerable.Range(-7, 15);
+
+            var moves = zeroes.Zip(minusSevenToSeven, (x,y)=>(x,y));
+            moves = moves.Union(minusSevenToSeven.Zip(zeroes, (x, y) => (x, y)));
+
+            return ApplyMoves(moves, board);
+        }
+
+        protected IEnumerable<Square> GetAvailableDiagonalMoves(Board board)
+        {
+            // Produce a zipped set of diagonal directions (South West, South East, North West, North East)
+            var diagonalDirections = new List<(int northSouth, int eastWest)>(new[] { (-1, -1), (-1, 1), (1, -1), (1, 1) });
+
+            // Send diagonal lines out 8 squares in each diagonal direction
+            var moves = diagonalDirections.Select(directions =>
+            {
+                return Enumerable.Range(1, 7).Select(i => (
+                    directions.northSouth * i,
+                    directions.eastWest * i)
+                );
+            });
+
+            // Flatten from four lists of lists of squares to one list
+            var movesFlattened = moves.SelectMany(x => x).ToList();
+
+            return ApplyMoves(movesFlattened, board);
+        }
+
+        protected IEnumerable<Square> ApplyMoves(IEnumerable<(int northSouth, int eastWest)> moves, Board board)
```

Comment:
All this use of `northSouth` and `eastWest` makes it a bit hard to read. What about creating a `Move` class with `rowChange` and `colChange` properties? 🙂

Reply:
That's a good idea! Done that in 6c5cbf8 🙂